<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>951</width>
    <height>729</height>
   </rect>
  </property>
  <property name="sizeIncrement">
   <size>
    <width>0</width>
    <height>1</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>1</width>
    <height>1</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>C++ Cryptography Exploration</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizeIncrement">
       <size>
        <width>1</width>
        <height>1</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>1</width>
        <height>1</height>
       </size>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideNone</enum>
      </property>
      <widget class="QWidget" name="rsa_tab">
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>RSA</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="rsa_nlen_label">
              <property name="maximumSize">
               <size>
                <width>125</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Key Length</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="rsa_nlen_select">
              <item>
               <property name="text">
                <string>2048 Bits</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>3072 Bits</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>7680 Bits</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>15360 Bits</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="rsa_key_type_label">
              <property name="maximumSize">
               <size>
                <width>125</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Private Key Type</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="rsa_key_type_select">
              <item>
               <property name="text">
                <string>Standard Form</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Quintuple Form</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="gen_type_label">
              <property name="maximumSize">
               <size>
                <width>125</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Generation Type</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="gen_type_select">
              <item>
               <property name="text">
                <string>Probably Prime</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Probably Prime with Probable Auxilary Primes</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Probably Prime with Provable Auxilary Primes</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Provably Prime</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Provably Prime with Provable Auxilary Primes</string>
               </property>
              </item>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QLabel" name="last_dur_label">
              <property name="maximumSize">
               <size>
                <width>250</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Last Duration</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="rsa_key_gen_button">
              <property name="minimumSize">
               <size>
                <width>125</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>125</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Generate</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4"/>
          </item>
          <item>
           <widget class="QTabWidget" name="key_details_tabs">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>150</height>
             </size>
            </property>
            <property name="tabPosition">
             <enum>QTabWidget::North</enum>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tab_n">
             <attribute name="title">
              <string>'n'</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QPlainTextEdit" name="n_text">
                <property name="autoFillBackground">
                 <bool>true</bool>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_d">
             <attribute name="title">
              <string>'d'</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QPlainTextEdit" name="d_text">
                <property name="autoFillBackground">
                 <bool>true</bool>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_e">
             <attribute name="title">
              <string>'e'</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QPlainTextEdit" name="e_text">
                <property name="autoFillBackground">
                 <bool>true</bool>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_p">
             <attribute name="title">
              <string>'p'</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QPlainTextEdit" name="text_p">
                <property name="autoFillBackground">
                 <bool>true</bool>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_q">
             <attribute name="title">
              <string>'q'</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QPlainTextEdit" name="text_q">
                <property name="autoFillBackground">
                 <bool>true</bool>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_dP">
             <attribute name="title">
              <string>'dP'</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <widget class="QPlainTextEdit" name="text_dP">
                <property name="autoFillBackground">
                 <bool>true</bool>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_dQ">
             <attribute name="title">
              <string>'dQ'</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <widget class="QPlainTextEdit" name="text_dQ">
                <property name="autoFillBackground">
                 <bool>true</bool>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_qInv">
             <attribute name="title">
              <string>'qInv'</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_11">
              <item>
               <widget class="QPlainTextEdit" name="text_qInv">
                <property name="autoFillBackground">
                 <bool>true</bool>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="documentTitle">
                 <string/>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="rsa_in_label">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Hex Input</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="rsa_in_text"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QPushButton" name="rsa_encrypt_button">
              <property name="text">
               <string>Encrypt</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="rsa_swap_out_button">
              <property name="text">
               <string>Move Output To Input</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="rsa_decrypt_button">
              <property name="text">
               <string>Decrypt</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="rsa_out_label">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Hex Output</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="rsa_out_text">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="sha_tab">
       <attribute name="title">
        <string>SHA</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_12">
        <item>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="hash_alg_label">
            <property name="text">
             <string>Hash Algorithm</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="hash_alg_select">
            <item>
             <property name="text">
              <string>SHA1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>SHA224</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>SHA256</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>SHA382</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>SHA512</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>SHA512/224</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>SHA512/256</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="hash_in_label">
            <property name="text">
             <string>Input</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPlainTextEdit" name="hash_in_text"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="digest_label">
            <property name="text">
             <string>Hash Digest</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QTextEdit" name="digest_text">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QPushButton" name="hash_button">
            <property name="text">
             <string>Hash</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
